dynamicio/mixins/with_kafka.py

```diff
@@ -76,9 +73,174 @@ class WithKafka:
     schema: DataframeSchema
     options: MutableMapping[str, Any]
     __kafka_config: Optional[Mapping] = None
-    __producer: Optional[KafkaProducer] = None
+    __producer: Optional[Producer] = None
     __key_generator: Optional[Callable[[Any, Mapping[Any, Any]], Optional[str]]] = None
     __document_transformer: Optional[Callable[[Mapping[Any, Any]], Mapping[Any, Any]]] = None
+    __key_serializer: Optional[Callable[[Optional[str]], Optional[bytes]]] = None
+    __value_serializer: Optional[Callable[[Mapping[Any, Any]], bytes]] = None
+
+    # N.B.: Please refer to https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md and update this config in case of a major release change.
+    VALID_CONFIG_KEYS = {
```

Comment:
I wouldn't check this stuff. Just pass it onto kafka and it should handle however it handles these.

Reply:
Nvm I forgot the "put everything into a dictionary then fish out stuff" pattern.

Reply:
yes... one of the things we should change for sure once we migrate away from the mixins pattern